### ISSUE TITLE
Build broken with aeson-1.4.3.0

### ISSUE THREAD

**Issue by matil019** [kallisti-dev/hs-webdriver#155]
`aeson-1.4.3.0` added a re-export of `<?>` from `Data.Aeson`. It conflicts with `Data.Attoparsec.ByteString.Char8`:

```
src\Test\WebDriver\Firefox\Profile.hs:217:16: error:
    Ambiguous occurrence ‘<?>’
    It could refer to either ‘Data.Aeson.<?>’,
                             imported from ‘Data.Aeson’ at src\Test\WebDriver\Firefox\Profile.hs:30:1-17
                             (and originally defined in ‘aeson-1.4.3.0:Data.Aeson.Types.Internal’)
                          or ‘AP.<?>’,
                             imported from ‘Data.Attoparsec.ByteString.Char8’ at src\Test\WebDriver\Firefox\Profile.hs:32:1-45
                             (and originally defined in ‘Data.Attoparsec.Combinator’)
    |
217 |   k <- prefKey <?> "preference key"
    |                ^^^

src\Test\WebDriver\Firefox\Profile.hs:219:16: error:
    Ambiguous occurrence ‘<?>’
    It could refer to either ‘Data.Aeson.<?>’,
                             imported from ‘Data.Aeson’ at src\Test\WebDriver\Firefox\Profile.hs:30:1-17
                             (and originally defined in ‘aeson-1.4.3.0:Data.Aeson.Types.Internal’)
                          or ‘AP.<?>’,
                             imported from ‘Data.Attoparsec.ByteString.Char8’ at src\Test\WebDriver\Firefox\Profile.hs:32:1-45
                             (and originally defined in ‘Data.Attoparsec.Combinator’)
    |
219 |   v <- prefVal <?> "preference value"
```

For the time being, adding an upper bound to `aeson` resolves the issue:

```a.patch
diff --git a/webdriver.cabal b/webdriver.cabal
index 8e36e7b..7d4a1ca 100644
--- a/webdriver.cabal
+++ b/webdriver.cabal
@@ -44,7 +44,7 @@ Library
   if flag(developer)
     cpp-options: -DCABAL_BUILD_DEVELOPER
   build-depends:   base == 4.*
-                 , aeson >= 0.6.2.0
+                 , aeson >= 0.6.2.0 && < 1.4.3.0
                  , http-client >= 0.3
                  , http-types >= 0.8
                  , text >= 0.11.3
```

That said, I believe fixing the import declarations is better, but I haven't looked into the code.

**Comment by kallisti-dev** [comment]
I don't see any issue with just fixing the import declarations.

**Comment by matil019** [comment]
I understand. May I open a PR?

**Comment by kallisti-dev** [comment]
Yes please. It's a bit tricky if you want to get full backwards
compatibility. Not sure if GHC or Cabal have released any new features
beyond CPP macros to address these scenarios..

**Comment by kallisti-dev** [comment]
After looking at the imports a bit more closely, I think the best choice
would be to explicitly list all imports from Data.Aeson instead of
importing everything as we're currently doing.

**Comment by DanBurton** [comment]
Bump. I'm temporarily removing `webdriver` from the Stackage nightly builds in order to allow for `aeson-1.4.3.0` in the nightly build plan. Please feel free to ping me or open a PR on the stackage repo once a compatible version of `webdriver` has been released.

**Comment by kallisti-dev** [comment]
I've pushed a fix: 9db008a0a659869d5938ac20b8679556409820ce

The change is now part of release [0.9.0.1](https://hackage.haskell.org/package/webdriver-0.9.0.1)

**Comment by kallisti-dev** [comment]
@DanBurton
